services/terraformer/server/domain/utils/backend/backend.go

```diff
@@ -23,37 +27,60 @@ type Backend struct {
 }
 
 type templateData struct {
-	ProjectName string
-	ClusterName string
-	MinioURL    string
-	AccessKey   string
-	SecretKey   string
-	DynamoURL   string
-	DynamoTable string
+	ProjectName    string
+	ClusterName    string
+	MinioURL       string
+	MinioAccessKey string
+	MinioSecretKey string
+	DynamoURL      string
+	DynamoTable    string
+	S3Name         string
+	Region         string
+	AwsAccessKey   string
+	AwsSecretKey   string
 }
 
 // CreateTFFile creates backend.tf file into specified Directory.
 func (b Backend) CreateTFFile() error {
 	template := templateUtils.Templates{Directory: b.Directory}
 
-	tpl, err := templateUtils.LoadTemplate(templates.BackendTemplate)
-	if err != nil {
-		return fmt.Errorf("failed to load template file backend.tpl for %s : %w", b.ClusterName, err)
-	}
+	// If the EXTERNAL_S3_BUCKET env is used, use the external_backend.tpl file
+	if externalBucketName != "" {
+		tpl, err := templateUtils.LoadTemplate(templates.ExternalBackendTemplate)
+		if err != nil {
+			return fmt.Errorf("failed to load template file external_backend.tpl for %s : %w", b.ClusterName, err)
+		}
+		data := templateData{
+			ProjectName:  b.ProjectName,
+			ClusterName:  b.ClusterName,
+			S3Name:       externalBucketName,
+			Region:       region,
+			AwsAccessKey: awsAccessKey,
+			AwsSecretKey: awsSecretAccessKey,
+			DynamoTable:  dynamoTable,
```

Comment:
Why not also pass the DynamoURL ?

Reply:
When using external dynamodb, you only need the Dynamo table name + aws region. Ref: https://docs.aws.amazon.com/general/latest/gr/ddb.html